src/main/java/de/tum/in/www1/artemis/web/rest/GradingInstructionResource.java

```diff
@@ -0,0 +1,151 @@
+    private String applicationName;
+
+    private final GradingInstructionService gradingInstructionService;
+
+    private final ExerciseService exerciseService;
+
+    private final AuthorizationCheckService authCheckService;
+
+    public GradingInstructionResource(GradingInstructionService gradingInstructionService, ExerciseService exerciseService, AuthorizationCheckService authCheckService) {
+        this.gradingInstructionService = gradingInstructionService;
+        this.exerciseService = exerciseService;
+        this.authCheckService = authCheckService;
+    }
+
+    /**
+     * GET /exercises/:id : get the "id" exercise.
+     *
+     * @param exerciseId the id of the exercise to retrieve
+     * @return the ResponseEntity with status 200 (OK) and with body the exercise, or with status 404 (Not Found)
+     */
+    @GetMapping("/grading-instruction/{exerciseId}")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<List<GradingInstruction>> getGradingInstructionsByExerciseId(@PathVariable Long exerciseId) {
+        log.debug("REST request to get Exercise : {}", exerciseId);
+
+        Exercise exercise = exerciseService.findOne(exerciseId);
+
+        if (!authCheckService.isAtLeastTeachingAssistantForExercise(exercise)) {
+            return forbidden();
+        }
+        List<GradingInstruction> gradingInstructions = gradingInstructionService.findAllForExercise(exercise);
+        return ResponseEntity.ok(gradingInstructions);
+    }
+
+    /**
+     * POST /grading-instruction : Create a new gradingInstruction.
+     *
+     * @param gradingInstruction the gradingInstruction to create
+     * @return the ResponseEntity with status 201 (Created) and with body the new gradingInstruction, or with status 400 (Bad Request) if the gradingInstruction has already an ID
+     * @throws URISyntaxException if the Location URI syntax is incorrect
+     */
+    @PostMapping("/grading-instruction")
+    @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<GradingInstruction> createGradingInstruction(@RequestBody GradingInstruction gradingInstruction) throws URISyntaxException {
+        log.debug("REST request to save GradingInstruction : {}", gradingInstruction);
+        if (gradingInstruction.getId() != null) {
+            return ResponseEntity.badRequest()
+                    .headers(HeaderUtil.createFailureAlert(applicationName, true, ENTITY_NAME, "idexists", "A new gradingInstruction cannot already have an ID")).body(null);
+        }
+
+        // fetch exercise from database to make sure client didn't change groups
+        Exercise exercise = exerciseService.findOne(gradingInstruction.getExercise().getId());
+        if (exercise == null) {
+            return ResponseEntity.badRequest().headers(
+                    HeaderUtil.createFailureAlert(applicationName, true, ENTITY_NAME, "exerciseNotFound", "The exercise belonging to this grading instructor does not exist"))
```

Comment:
```suggestion
                    HeaderUtil.createFailureAlert(applicationName, true, ENTITY_NAME, "exerciseNotFound", "The exercise belonging to this grading instruction does not exist"))
```

Reply:
I am impressed by how you find these mistakes ^^